util.c

```diff
@@ -787,3 +787,65 @@ char *filter_printf(const char *filter, const char *user) {
   filter_result_len(filter, user, result);
   return result;
 }
+
+size_t rfc4515_length ( const char* in, char* out )
+{
+    const char* pos = NULL;
+    size_t result = 0;
+    char* specials = (char*)malloc( 4 * sizeof(char) );
+    specials[0] = 0x5c;
+    specials[1] = 0x2a;
+    specials[2] = 0x28;
+    specials[3] = 0x29;
+
+    do
+    {
+        pos = strpbrk( in, specials );
+
+        size_t len = 0;
+        if( pos != NULL )
+        {
+            len = pos - in;
+        }
+        else
+        {
+            len = strlen( in );
+        }
+
+        if( out != NULL )
+        {
+            strncpy( out, in, len );
+            out += len;
+        }
+
+        result += len;
+        in += len;
+
+        if( pos != NULL )
+        {
+            if( out != NULL )
+            {
+              *(out) = 0x5c;
+              strncpy( (out+1), pos, 1 );
```

Comment:
since this is just copying one byte, do direct assignment instead, `*out++ = pos` and direct increment on the line above so we can get rid of += 2 on the next line.

Reply:
I am not sure what do you mean. Do you mean ...

- change `*(out) = 0x5c;` to `*out++ = 0x5c;`
- change `strncpy( (out+1), pos, 1 );` to `*out++ = pos;`
- and remove `out += 2;`

Reply:
Done